### PR TITLE
Bump stdlib version to 0.1.0

### LINKED ISSUE #292: Is stdlib versioned?
While writing package files for stdlib (see #291) I encountered a quite simple issue what puzzled me: What is the version number of stdlib?

It turns out that some package files simply don't work if there is no version number specified, *i.e.* pkg-config will not use any pc-file without a Version entry. Not specifying any version is therefore not possible if I want to use stdlib in a non-CMake project. I could *just define* a dummy version number for my purpose, but this calls for trouble.

Maybe this issue has been discussed already somewhere, but I didn't found any hint on this here.

### PR DESCRIPTION
**Pull request by awvwgk**
Create first release for stdlib. The release notes will be build from the current changelogs entries, feel free to suggests additions to the changelog. After this patch is merged, a new release will be tagged with `v0.1.0`.

*Features, proposals, and fixes that should be included in the first release of the Fortran standard library (0.1.0) should be added to the milestone [v0.1.0](https://github.com/fortran-lang/stdlib/milestone/1).*

Closes #292 
~~Requires #537~~

### CODE REVIEW

**Review by jvdp1 — APPROVED** [review]
Nothing special. I agree with it.

**Comment by awvwgk** [comment]
There is now a release draft (visible to all maintainers) at https://github.com/fortran-lang/stdlib/releases, you can edit the draft and save the changes there with the *save draft* button (note that release drafts don't have a permanent URL, therefore I can't link it from here).

**Comment by jvdp1** [comment]
Thank you @awvwgk for this work.

while I was checking the links for #272, I was wondering if this version number should also appear in `stdlib.fortran-lang.org`. `CHANGELOG.md` might be already consider by `ford`. However I didn't check it. Probably good to check this at the end of this PR.

**Comment by zoziha** [comment]
@awvwgk `Linspace`, `logspace`, and `arange` should belong to `stdlib_math`.
Thanks, this PR is really a milestone.
Maybe someday I can install fortran `stdlib` on `msys2` :)
image

**Review by zoziha — APPROVED** [review]

File: CHANGELOG.md
Comment (zoziha) [inline CHANGELOG.md:26]: `outer_product` should belong to `stdlib_linalg`.

File: CHANGELOG.md
Comment (zoziha) [inline CHANGELOG.md:42]: Two days ago, the `gcd` function was also added to `stdlib_math`. (`gcd`, see https://github.com/fortran-lang/stdlib/pull/539)

Now `stdlib_math` has: `clip, linspace, logspace, arange, gcd`.

**Comment by awvwgk** [comment]
>  Maybe someday we can install fortran stdlib on msys2 :)

I can setup a PKGBUILD once this is merged. Packaging for msys2 is quite straight-forward, you can try it yourself by adjusting fpm-git PKGBUILD given in https://github.com/fortran-lang/fpm/wiki/Packaging-Fpm#msys2-windows for CMake.

**Comment by Jim-215-Fisher** [comment]
Finally, we have a release version. Regarding to stdlib_stats_distribution_PRNG module, I am not sure whether it should be in stats_distribution package. GSL has a separate `random numbers` package in addition to `random distributions` and `statistics`.

**Comment by awvwgk** [comment]
@Jim-215-Fisher I opened #544 to discuss the change regarding the PRNG distribution module.

**Comment by awvwgk** [comment]
I'll go ahead and merge this PR and publish the first release of stdlib. Thanks everybody for the hard work so far.

**Comment by jvdp1** [comment]
Thank you @awvwgk for all your work to make this first release!

**Pull request merged**
